libsmack/sys/smack.h

```diff
@@ -340,6 +341,34 @@ int smack_load_policy(void);
  */
 int smack_set_relabel_self(const char **labels, int cnt);
 
+/*!
+ * Set the list of labels that will be allowed to have effective CAP_MAC_ADMIN
+ * and CAP_MAC_OVERRIDE. This set of labels will be applied (written)
+ * to the kernel interface "onlycap". Setting empty list causes CAP_MAC_ADMIN &
+ * CAP_MAC_OVERRIDE to be unconstrained by any specific Smack label. Empty
+ * list of onlycap Smack labels is the default kernel configuration. The caller
+ * must have CAP_MAC_ADMIN capability. Caller may effectively loose
+ * the capability after successfull return from the function if its Smack label
+ * is not on the list of labels.
+ *
+ * @param labels list of labels (NULL for empty list)
+ * @param cnt number of labels (0 for empty list)
+ * @return Returns 0 on success and negative value on failure
+ *
+ */
+int smack_set_onlycap(const char **labels, int cnt);
+
+
+/*!
+ * Set the list of labels that will be allowed to have effective CAP_MAC_ADMIN
+ * and CAP_MAC_OVERRIDE. This function reads list of labels from file path
+ * passed as argument and calls smack_set_onlycap afterwards.
+ *
+ * @param path path to file containing list of onlycap Smack labels
+ * @return Returns 0 on success and negative value on failure
+ */
+int smack_set_onlycap_from_file(const char* path);
```

Comment:
Whether we like it or not, there is an established convention in this library that functions named like `smack_*_from_file` take file descriptor as an argument. Don't break that convention.

By the way, `const char* path` should be written as `const char *path` (libsmack is written in kernel-like code style).